src/Primitives.hs

```diff
@@ -221,53 +217,63 @@ primitiveImplements x@(XObj _ _ _) ctx args =
 
 define :: Bool -> Context -> XObj -> IO Context
 define hidden ctx@(Context globalEnv _ typeEnv _ proj _ _ _) annXObj =
-  let previousType =
-        case lookupInEnv (getPath annXObj) globalEnv of
-          Just (_, Binder _ found) -> xobjTy found
-          Nothing -> Nothing
-      previousMeta = existingMeta globalEnv annXObj
-      adjustedMeta =
-        if hidden
-          then Meta.set "hidden" trueXObj previousMeta
-          else previousMeta
-   in case annXObj of
-        XObj (Lst (XObj (Defalias _) _ _ : _)) _ _ ->
-          pure (ctx {contextTypeEnv = TypeEnv (envInsertAt (getTypeEnv typeEnv) (getPath annXObj) (Binder adjustedMeta annXObj))})
-        XObj (Lst (XObj (Deftype _) _ _ : _)) _ _ ->
-          pure (ctx {contextTypeEnv = TypeEnv (envInsertAt (getTypeEnv typeEnv) (getPath annXObj) (Binder adjustedMeta annXObj))})
-        XObj (Lst (XObj (DefSumtype _) _ _ : _)) _ _ ->
-          pure (ctx {contextTypeEnv = TypeEnv (envInsertAt (getTypeEnv typeEnv) (getPath annXObj) (Binder adjustedMeta annXObj))})
-        _ ->
-          do
-            when (projectEchoC proj) $
-              putStrLn (toC All (Binder emptyMeta annXObj))
-            case previousType of
-              Just previousTypeUnwrapped ->
-                unless (areUnifiable (forceTy annXObj) previousTypeUnwrapped) $
-                  do
-                    emitWarning
-                      ( "Definition at " ++ prettyInfoFromXObj annXObj ++ " changed type of '" ++ show (getPath annXObj)
-                          ++ "' from "
-                          ++ show previousTypeUnwrapped
-                          ++ " to "
-                          ++ show (forceTy annXObj)
-                      )
-              Nothing -> pure ()
-            case Meta.get "implements" previousMeta of
-              Just (XObj (Lst interfaces) _ _) ->
-                do
-                  let result = foldM (\ctx' (xobj, interface) -> registerInInterface ctx' xobj interface) ctx (zip (cycle [annXObj]) (map getPath interfaces))
-                  case result of
-                    Left err ->
-                      do
-                        case contextExecMode ctx of
-                          Check ->
-                            let fppl = projectFilePathPrintLength (contextProj ctx)
-                             in putStrLn (machineReadableInfoFromXObj fppl annXObj ++ " " ++ err)
-                          _ -> putStrLnWithColor Red err
-                        pure ctx
-                    Right ctx' -> pure (ctx' {contextGlobalEnv = envInsertAt globalEnv (getPath annXObj) (Binder adjustedMeta annXObj)})
-              _ -> pure (ctx {contextGlobalEnv = envInsertAt globalEnv (getPath annXObj) (Binder adjustedMeta annXObj)})
+  pure (hideIt freshBinder)
+    >>= \newBinder -> case annXObj of
+      XObj (Lst (XObj (Defalias _) _ _ : _)) _ _ ->
+        pure (ctx {contextTypeEnv = TypeEnv (envInsertAt (getTypeEnv typeEnv) (getPath annXObj) newBinder)})
```

Comment:
Since all these three matches do the same thing, maybe wrap it up in a function called `defineInTypeEnv` or something? `defineGlobal` could maybe change to `defineInValueEnv` or something to mirror it..?

Reply:
Sounds good to me!